packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -160,14 +161,17 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
     @required AxisDirection crossAxisDirection,
     @required ViewportOffset offset,
     double cacheExtent,
+    bool autoCache = false,
```

Comment:
should we assert that when you set autoCache to true, cacheExtent is null to make the contract that it gets ignored explicit?

Reply:
We can't - the cache extent is still used but the widget can no longer control it.

Reply:
Trying a different approach here...